protected-api-access-protected-api/app.js

```diff
@@ -0,0 +1,81 @@
+/*
+ This application demonstrates how to issue a call to a protected API
+ using the on-behalf-of OAuth2 flow.  A request will be issued to the
+ Microsoft Graph /me endpoint on behalf of the authenticated user.
+*/
+
+// Microsoft Authentication Library (MSAL) for Node.js
+const msal = require('@azure/msal-node')
+
+// Node.js Express Framework
+const express = require('express')
+
+// Used to validate JWT access tokens
+const jwt = require('express-jwt')
+const jwks = require('jwks-rsa')
+const jwtAuthz = require('express-jwt-authz')
+
+// Used to make the HTTP GET request to the Graph API
+const https = require('https')
+
+// MSAL configs
+const config = {
+  auth: {
+    // 'Application (client) ID' of app registration in Azure portal - this value is a GUID
+    clientId: '',
+   
+    // Client secret 'Value' (not the ID) from 'Client secrets' in app registration in Azure portal
+    clientSecret: '',
+   
+    // Full directory URL, in the form of https://login.microsoftonline.com/<tenant>
+    authority: ''
+  }
+}
+
+
+// Initialize MSAL
+const msalConfidentialClientApp = new msal.ConfidentialClientApplication(config)
+
+// Initialize Express
+const app = express()
+
+// Add Express middleware to validate JWT access tokens
+app.use(jwt({
+  secret: jwks.expressJwtSecret({
+    jwksUri: config.auth.authority + '/discovery/v2.0/keys'
+  }),
+  audience: config.auth.clientId,
+  issuer: config.auth.authority + '/v2.0',
+  algorithms: ['RS256']
+}))
+
+// Allow access to the /me endpoint if the provided JWT access token has
+// the 'user_impersonation' scope.
+app.get('/me', jwtAuthz(['user_impersonation'], { customScopeKey: 'scp' }), (req, res) => {
+  // Get the user's token
+  const authHeader = req.headers.authorization
+
+  // Required for the on-behalf-of request (token and scope(s))
+  const oboRequest = {
+    oboAssertion: authHeader.split(' ')[1],
+    scopes: ['user.read']
+  }
+
+  // Send the user's token to Graph, to receive a token for Graph on-behalf-of the user
+  msalConfidentialClientApp.acquireTokenOnBehalfOf(oboRequest).then((response) => {
+    const options = {
+      method: 'GET',
+      headers: { Authorization: `Bearer ${response.accessToken}` }
+    }
+
+    // Perform an HTTP GET request against the Graph endpoint with the token issued by Graph on-behalf-of the user
```

Comment:
Just a quick typo I saw, Graph doesn't issue tokens.

```suggestion
    // Perform an HTTP GET request against the Graph endpoint with the access token issued by
    // Azure AD on behalf of the user.
```